crates/bitwarden-vault/src/cipher/cipher.rs

```diff
@@ -455,6 +454,24 @@ impl Cipher {
             .map(|kind| kind.get_copyable_fields(Some(self)))
             .unwrap_or_default()
     }
+
+    /// Extracts and sets the CipherType-specific fields from the opaque `data` field.
+    pub(crate) fn populate_cipher_types(&mut self) {
+        let Ok(data) =
+            serde_json::from_str::<serde_json::Value>(self.data.as_deref().unwrap_or("{}"))
```

Comment:
❓ looks like we are doing a double deserialization. Would it be better if we did this directly on the target type